ci-operator/step-registry/aws/provision/route53/private-hosted-zone/associate-diff-vpc/aws-provision-route53-private-hosted-zone-associate-diff-vpc-commands.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+set -x
+
+REGION="${LEASED_RESOURCE}"
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+if [[ ${BASE_DOMAIN} == "" ]]; then
+  echo "No base_domain provided, exit now."
+  exit 1
+fi
+ROUTE53_HOSTED_ZONE_NAME="${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+VPC_ID=$(cat "${SHARED_DIR}/vpc_id")
+# Use a timestamp to ensure the caller reference is unique, as we've found
+# cluster name can get reused in specific situations.
+TIMESTAMP=$(date +%s)
+CALLER_REFERENCE_STR="${ROUTE53_HOSTED_ZONE_NAME}-${TIMESTAMP}"
+
+
+# Create PHZ
+echo "AWS Account: using shared account"
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred_shared_account"
+
+echo "Creating PHZ ..."
+# default VPC is only for create *PRIVATE* hosted zone.
+vpc_for_phz_creation=$(aws --region ${REGION} ec2 describe-vpcs | jq -r '.Vpcs[] | select(.IsDefault==true) | .VpcId')
+if [[ ${vpc_for_phz_creation} == "" ]] || [[ ${vpc_for_phz_creation} == "null" ]]; then
+  echo "No default VPC found in shared account, can not create private hosted zone, exit now"
+  exit 1
+fi
+
+HOSTED_ZONE_CREATION=$(aws --region ${REGION} route53 create-hosted-zone --name "${ROUTE53_HOSTED_ZONE_NAME}" --caller-reference "${CALLER_REFERENCE_STR}" --vpc VPCRegion=${REGION},VPCId=${vpc_for_phz_creation})
+HOSTED_ZONE_ID="$(echo "${HOSTED_ZONE_CREATION}" | jq -r '.HostedZone.Id' | awk -F / '{printf $3}')"
+echo "${HOSTED_ZONE_ID}" > "${SHARED_DIR}/hosted_zone_id"
+
+CHANGE_ID="$(echo "${HOSTED_ZONE_CREATION}" | jq -r '.ChangeInfo.Id' | awk -F / '{printf $3}')"
+aws --region "${REGION}" route53 wait resource-record-sets-changed --id "${CHANGE_ID}" &
+wait "$!"
+echo "Hosted zone ${HOSTED_ZONE_ID} successfully created."
+
+
+echo "AWS Account: using shared account"
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred_shared_account"
```

Comment:
Seem like we did not switched account since line 27, so we do not need this line?

Reply:
Updated.